content/python/concepts/generators/generators.md

```diff
@@ -0,0 +1,60 @@
+---
+Title: 'Generators'
+Description: 'In Python, a generator is a function or expression that will process a given iterable one object at a time, on demand.'
+Subjects:
+  - 'Computer Science'
+  - 'Data Science'
+Tags:
+  - 'Generators'
+  - 'OOP'
+  - 'Loops'
+CatalogContent:
+  - 'learn-python-3'
+  - 'paths/data-science'
+---
+
+In Python, a generator is a function or expression that will process a given iterable one object at a time, on demand. A generator will return a generator object, which can be cast as a list or some other data structure as appropriate.
+
+## Generators
+
+Generators are a convenient means of employing [iterator](https://www.codecademy.com/resources/docs/python/iterators) functionality within the syntax of a function or expression. One of the main advantages of generators is that they evaluate items on demand, which means only one item is in memory at a time in lieu of the entire sequence (as with a list).
+
+## A Generator Function
```

Comment:
```suggestion
### A Generator Function
```

It seems like this section and the other one, "A Generator Expression", sort of fall under the "Generators" section, no?

Reply:
I created to the subheads for the sake of breaking up the content and grouping the details of these two classes of generator implementations. I don't think it makes sense to have just one section from a readability and a content perspective

Reply:
I agree, it wouldn't make sense to lump all of this into one section.

> It seems like this section and the other one, "A Generator Expression", sort of fall under the "Generators" section, no?

This was meant more in the spirit of supporting changing to level 3 headers.